### PR TITLE
[DOC] Semi-SL for multi-class classification algorithm description part

### PR DESCRIPTION
**Pull request by harimkang**
## Summary
Semi-SL for multi-class classification Documentation part

image

### CODE REVIEW

**Review by JihwanEom — APPROVED** [review]

**Review by sungmanc — APPROVED** [review]

File: docs/source/explanation/algorithms/classification/multi_class_classification.rst [hunk 1]
```diff
@@ -98,7 +98,60 @@ In the table below the top-1 accuracy on some academic datasets using our :ref:`
 Semi-supervised Learning
 ************************
 
-To be added soon
+Semi-SL (Semi supervised Learning) is a type of machine learning algorithm that uses both labeled and unlabeled data to improve the performance of the model. This is particularly useful when labeled data is limited, expensive or time-consuming to obtain.
```
Comment (kprokofi) [inline docs/source/explanation/algorithms/classification/multi_class_classification.rst:101]: We already described the definition of the Semi-SL task in the algorithms/index file

File: docs/source/explanation/algorithms/classification/multi_class_classification.rst [hunk 2]
```diff
@@ -98,7 +98,60 @@ In the table below the top-1 accuracy on some academic datasets using our :ref:`
...
+
+`FixMatch <https://arxiv.org/abs/2001.07685>`_ is a specific implementation of Semi-SL that has been shown to be effective in various applications. FixMatch introduces pseudo-labeling, which is the process of generating labels for the unlabeled data and treating them as if they were labeled data. Pseudo-labeling is based on the idea that the model's prediction for the unlabeled data is likely to be correct, which can improve the model's accuracy and reduce the need for labeled data.
```
Comment (kprokofi) [inline docs/source/explanation/algorithms/classification/multi_class_classification.rst:103]: We use `FixMatch <https://arxiv.org/abs/2001.07685>`_  as a core algorithm for Semi-SL task solving. It is a specific ....

File: docs/source/explanation/algorithms/classification/multi_class_classification.rst [hunk 3]
```diff
@@ -98,7 +98,60 @@ In the table below the top-1 accuracy on some academic datasets using our :ref:`
...
+
+In Semi-SL, the pseudo-labeling process is combined with a consistency loss that ensures that the predictions of the model are consistent across augmented versions of the same data. This helps to reduce the impact of noisy or incorrect labels that may arise from the pseudo-labeling process. Additionally, Our algorithm uses a combination of strong data augmentations and a specific optimizer called Sharpness-Aware Minimization (SAM) to further improve the accuracy of the model.
```
Comment (kprokofi) [inline docs/source/explanation/algorithms/classification/multi_class_classification.rst:105]: Additionally, our algorithm ... (lower case)

File: docs/source/explanation/algorithms/classification/multi_class_classification.rst [hunk 4]
```diff
@@ -98,7 +98,60 @@ In the table below the top-1 accuracy on some academic datasets using our :ref:`
...
+
+Overall, OTX Semi-SL are powerful techniques for improving the performance of machine learning models with limited labeled data. They can be particularly useful in domains where labeled data is expensive or difficult to obtain, and can help to reduce the time and cost associated with collecting labeled data.
```
Comment (kprokofi) [inline docs/source/explanation/algorithms/classification/multi_class_classification.rst:107]: Overall, OTX utilizes powerful techniques for improving the performance of Semi-SL algorithm with limited ....

File: docs/source/explanation/algorithms/classification/multi_class_classification.rst [hunk 5]
```diff
@@ -98,7 +98,60 @@ In the table below the top-1 accuracy on some academic datasets using our :ref:`
...
+
+- ``Pseudo-labeling (FixMatch)``: A specific implementation of Semi-SL that combines the use of pseudo-labeling with a consistency loss, strong data augmentations, and a specific optimizer called Sharpness-Aware Minimization (SAM) to improve the performance of the model.
+
+- ``Adaptable Threshold``: A novel addition to our solution that calculates a class-wise threshold for pseudo-labeling, which can solve the issue of imbalanced data and produce high-quality pseudo-labels that improve the overall score.
+
+- ``Unlabeled Warm Up Loss``: A technique for preventing the initial unstable learning of pseudo-labeling by increasing the coefficient of the unlabeled loss from 0 to 1.
+
+- ``Exponential Moving Average (EMA)``: A technique for maintaining a moving average of the model's parameters, which can improve the generalization performance of the model.
+
+- ``Other solutions``: Other than that, we use several solutions that apply to supervised learning (No bias Decay, Augmentations, etc.).
```
Comment (kprokofi) [inline docs/source/explanation/algorithms/classification/multi_class_classification.rst:119]: ``Additional techniques`` or `Additional training tricks` - may sound better

File: docs/source/explanation/algorithms/classification/multi_class_classification.rst [hunk 6]
```diff
@@ -98,7 +98,60 @@ In the table below the top-1 accuracy on some academic datasets using our :ref:`
...
+
+Please, refer to the :doc:`tutorial <../../../tutorials/advanced/semi_sl>` how to train semi supervised learning. Based on MobileNet-V3-large-1x, it takes about 3 times longer than conventional supervised learning.
```
Comment (kprokofi) [inline docs/source/explanation/algorithms/classification/multi_class_classification.rst:121]: semi-supervised

Comment (kprokofi) [inline docs/source/explanation/algorithms/classification/multi_class_classification.rst:121]: "Training time depends on the number of unlabeled images and can be up to several times longer then conventional supervised learning"

File: docs/source/explanation/algorithms/classification/multi_class_classification.rst [hunk 7]
```diff
@@ -98,7 +98,60 @@ In the table below the top-1 accuracy on some academic datasets using our :ref:`
...
+
+In the table below the top-1 accuracy on some academic datasets using our pipeline is presented. Same as the supervised setting except for an image for unlabeled and an additional batch size.
+
+4 images per class (+ unlabeled image for Semi-SL)
```
Comment (kprokofi) [inline docs/source/explanation/algorithms/classification/multi_class_classification.rst:125]: just 1 unlabeled image? It is strange setting... I would rather present larger datasets with more unlabeled data.
Lets leave it as is for now, but I prefer to change validation protocol with larger datasets in the future

Comment (harimkang) [reply docs/source/explanation/algorithms/classification/multi_class_classification.rst:125]: This does not mean 1 unlabeled image, but many unlabeled images. I will correct the word.

**Pull request merged**